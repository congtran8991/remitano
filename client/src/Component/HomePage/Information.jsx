import React from "react";

const Information = () => {
  return (
    <div className="information-video">
      <div className="title-video">Title Video</div>
      <div className="mail-user">Share by: congtran8991@gmail.com</div>
      <div className="description">
        <div className="label">Description</div>
        <div className="description-content">
        I think that the error message is a little bit misleading, as it has
          nothing to do with your actual host, but is more about how resources
          from youtube.com are referenced on the page. There are two things I
          would suggest in order to get rid of this error message. (At least
          these were working in my case.)
        </div>
      </div>
    </div>
  );
};

export default Information;
