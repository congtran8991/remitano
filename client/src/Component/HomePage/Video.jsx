import React, { useRef } from "react";
import ReactPlayer from "react-player";

const Video = (props) => {
  const { url } = props;
  const playerRef = useRef(null);
  return (
    <div className="video">
      <ReactPlayer
        controls
        style={{
          maxWidth: "37rem",
          minWidth: "20rem",
          maxHeight: "14.5rem",
          minHeight: "2.5rem",
        }}
        width={"100%"}
        height={"300px"}
        url={url}
        ref={playerRef}
      />
    </div>
  );
};

export default Video;
