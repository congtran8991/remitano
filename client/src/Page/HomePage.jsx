import React, { useEffect, useState } from "react";
import Video from "../Component/HomePage/Video";
import Information from "../Component/HomePage/Information";
import { AXIOS } from "../Utils/axios";

const HomePage = () => {
  const [listDataMovie, setListDataMovie] = useState([]);
  useEffect(() => {
    getListMovieData();
  }, []);
  const getListMovieData = async () => {
    try {
      const param = {
        path: "/movie/listMovie",
        method: "GET",
      };
      let dataApi = await AXIOS(param);
      setListDataMovie(dataApi.data);
    } catch (error) {
      setListDataMovie([]);
    }
  };
  return (
    <div className="home-page">
      <div className="list-video">
        {listDataMovie.map((item) => {
          return (
            <div key={item._id} className="row-video">
              <Video url={item.url} />
              <Information email={item.user.email}/>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default HomePage;
