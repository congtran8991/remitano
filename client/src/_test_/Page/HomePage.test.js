/* eslint-disable testing-library/no-unnecessary-act */
import { act, render } from "@testing-library/react";
import HomePage from "../../Page/HomePage";
const { AXIOS } = require("../../Utils/axios");
const dataSuccess = {
  success: true,
  message: "Lấy danh sách thành công",
  data: [
    {
      url: "https://www.youtube.com/watch?v=YYmzj5DK_5s",
      user: {
        userId: "643ffe24b2251e31ada3e4e7",
        email: "daniel.nguyen@inspireventures.com",
      },
    },
  ],
};

jest.mock("../../Utils/axios", () => ({
  AXIOS: jest.fn().mockResolvedValueOnce([]),
}));

jest.mock("../../Utils/axios", () => ({
  AXIOS: jest.fn().mockRejectedValueOnce({}),
}));

describe("HomePage", () => {
  test("Render HomePage", async () => {
    await act(async () => {
       render(<HomePage />);
    });
  });
  test("should return array data when axios get listMovie request success", async () => {
    const param = {
      path: "/movie/listMovie",
      method: "GET",
    };
    AXIOS.mockResolvedValueOnce(dataSuccess);
    let result = await AXIOS(param);
    expect(result).toEqual(dataSuccess);
  });
  test("should return array data when axios get listMovie request fail", async () => {
    const param = {
      path: "/movie/listMovie",
      method: "GET",
    };
    const dataFail = {
      success: false,
      message: "Lấy danh sách không thành công",
      data: [],
      err: new Error("Some error"),
    };
    AXIOS.mockRejectedValueOnce(dataFail);
    act(async () => {
      await expect(AXIOS(param)).rejects.toEqual(dataFail);
    });
  });
});
