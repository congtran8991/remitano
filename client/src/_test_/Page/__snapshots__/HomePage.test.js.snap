// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Render HomePage 1`] = `undefined`;

exports[`HomePage renders component HomePage 1`] = `undefined`;

exports[`HomePage renders component HomePage 2`] = `
<div>
  <div
    class="home-page"
  >
    <div
      class="list-video"
    />
  </div>
</div>
`;
